Entities/CrystalBombDetonator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Celeste.Mod.Entities;
+using Microsoft.Xna.Framework;
+using Monocle;
+
+namespace Celeste.Mod.SpringCollab2020.Entities {
+    [Tracked(false)]
+    [CustomEntity("SpringCollab2020/crystalBombDetonator")]
+    public class CrystalBombDetonator : Solid {
+        // Duration of flashing anim
+        public float Flash = 0f;
+
+        // Used in the shape of the "wave" on the renderer
+        public float Solidify = 0f;
+        private float solidifyDelay = 0f;
+
+        // Used to propagate state
+        private List<CrystalBombDetonator> adjacent = new List<CrystalBombDetonator>();
+
+        // If the field is currently in its flashing state (used to propagate Flashing state to adjacent fields)
+        public bool Flashing = false;
+
+        // Particle array and a list of speeds
+        private List<Vector2> particles = new List<Vector2>();
+        private readonly static float[] speeds = new float[] {
+            12f,
+            20f,
+            40f
+        };
+
+        // Cached method reference, first calculated in Update
```

Comment:
```suggestion
        // Cached method reference, first calculated in CheckForBombs
```

Reply:
kids, don't write code comments while on cough syrup

i'll note that for another tidy pass later